se/se_epub.py

```diff
@@ -415,9 +418,9 @@ def recompose(self) -> str:
 
 				output_xhtml = regex.sub(fr"<img.+?src=\"\.\./images/{match}\.svg\".*?/>", svg, output_xhtml)
 
-		with tempfile.NamedTemporaryFile(mode="w+", delete=False) as file:
-			file.write(output_xhtml)
-			file_name = Path(file.name)
+		with tempfile.NamedTemporaryFile(mode="w+", delete=False) as temp_file:
```

Comment:
Let's leave this as `file`, which is consistent with the rest of the codebase. Same with `binary_file` on line 529.

Reply:
The type of the variable I named `temp_file` is `IO[str]` but when `file` is used earlier (and a few lines later), the type of `file` is `TextIO`. The types are related -- `IO[str]` is actually the parent class of `TextIO` -- but because `TextIO` is a more specific type, the temp file value cannot be assigned.

Similarly, the variable that I renamed to `binary_file` has a type of `BinaryIO` and cannot be assigned to a `file` variable that has the type `TextIO`.

See https://docs.python.org/3/library/typing.html#typing.IO for a description of the IO types.

If you want to standardize on naming, I would suggest having one standard name for each type rather than trying to apply one name to all IO types.

Reply:
I see, so it's more of a mypy issue. From my perspective, if the variable is being declared within the scope of the block, then I don't see why we can't declare a new block later on with the same variable name, but of a different type. The names are the same but the blocks and their scopes are independent. Presumably a variable declared in a block scope like these are, cannot be used again after the block ends. Unless that's another unexpected Python quirk?

I like just `file` because nobody really cares if it's a temp or a binary or whatever file most of the time, and those distinctions become clear once the code is read anyway. It's naming a variable what it *is*, not what it's *type* is--why we write `counter` instead of `intCounter` like we did back in the days before IDEs existed.

But I suppose that's all moot if mypy complains anyway. Can't say I'm a huge fan of how Python has jerry-rigged its typing system :) 

Anyway thanks for working on this, this has been a huge help!

Reply:
Unlike some other languages, the Python language does not create scopes for statements like `if`, `for`, and `with`. A variable declared within one of those blocks is available in that function scope from that point on. So, yes, another Python quirk.

If you want to limit the scope of variables in Python, you pretty much have to write smaller functions. I think that's one of the reasons pylint has a too-many-statements check. As functions get longer, it gets more difficult to keep track of all the local variables and avoid conflicts.